src/bots/userPnlSettler.ts

```diff
@@ -63,27 +71,73 @@ const errorCodesToSuppress = [
 	6259, // Error Code: NoUnsettledPnl. Error Number: 6259. Error Message: NoUnsettledPnl.
 ];
 
+// =============================================================================
+// TYPES
+// =============================================================================
+
+interface UserToSettle {
+	settleeUserAccountPublicKey: PublicKey;
+	settleeUserAccount: UserAccount;
+	pnl: number;
+}
+
+interface MarketDataCache {
+	perpMarketAndOracleData: {
+		[marketIndex: number]: {
+			marketAccount: PerpMarketAccount;
+			oraclePriceData: OraclePriceData;
+		};
+	};
+	spotMarketAndOracleData: {
+		[marketIndex: number]: {
+			marketAccount: SpotMarketAccount;
+			oraclePriceData: OraclePriceData;
+		};
+	};
+}
```

Comment:
kinda nit: i dont actually know why we had this to cache market+oracle accounts, since the exact same data is essentially 'cached' at the driftClient level, you can probably get rid of this if you want.

honestly it might be better to not have this since you want the most recent data (from driftclient) to have accurate pnl pool values etc.